pkg/api/datasources.go

```diff
@@ -92,12 +92,7 @@ func (hs *HTTPServer) GetDataSourceById(c *models.ReqContext) response.Response
 		return response.Error(500, "Failed to query datasources", err)
 	}
 
-	filtered, err := hs.filterDatasourcesByQueryPermission(c.Req.Context(), c.SignedInUser, []*models.DataSource{query.Result})
```

Comment:
If we remove `filterDatasourcesByQueryPermission`, are we going to have a problem if `rbacEnabled` is `false`?

Reply:
No then the fallback handler will be called for the endpoints (reqOrgAdmin)